File: training-data/csharp/FloydWarshall.cs
using System;

class FloydWarshall
{
    static int[,] Solve(int numNodes, (int, int, int)[] edges)
    {
        const int inf = int.MaxValue / 2;
        var dist = new int[numNodes, numNodes];
        for (int i = 0; i < numNodes; i++)
            for (int j = 0; j < numNodes; j++)
                dist[i, j] = i == j ? 0 : inf;
        foreach (var (u, v, w) in edges) dist[u, v] = w;

        for (int k = 0; k < numNodes; k++)
            for (int i = 0; i < numNodes; i++)
                for (int j = 0; j < numNodes; j++)
                    if (dist[i, k] + dist[k, j] < dist[i, j]) dist[i, j] = dist[i, k] + dist[k, j];

        return dist;
    }

    static void Main()
    {
        var edges = new[] { (0, 1, 3), (0, 2, 8), (1, 2, 2), (2, 3, 1), (3, 0, 4) };
        var dist = Solve(4, edges);
        for (int i = 0; i < 4; i++)
        {
            for (int j = 0; j < 4; j++) Console.Write(dist[i, j] + " ");
            Console.WriteLine();
        }
    }
}

File: training-data/ocaml/floyd_warshall.ml
let inf = 1000000

let floyd_warshall n edges =
  let dist = Array.make_matrix n n inf in
  for i = 0 to n - 1 do
    dist.(i).(i) <- 0
  done;
  List.iter (fun (u, v, w) -> dist.(u).(v) <- w) edges;
  for k = 0 to n - 1 do
    for i = 0 to n - 1 do
      for j = 0 to n - 1 do
        if dist.(i).(k) + dist.(k).(j) < dist.(i).(j) then
          dist.(i).(j) <- dist.(i).(k) + dist.(k).(j)
      done
    done
  done;
  dist

let () =
  let edges = [ (0, 1, 3); (0, 2, 8); (1, 2, 2); (2, 0, 5); (2, 3, 1); (3, 1, 4) ] in
  let dist = floyd_warshall 4 edges in
  Array.iter
    (fun row ->
      Array.iter (fun d -> Printf.printf "%d " d) row;
      print_newline ())
    dist
